src/dali_executor/request_id.h

```diff
@@ -0,0 +1,66 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2020-2023 NVIDIA CORPORATION & AFFILIATES
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef DALI_BACKEND_DALI_EXECUTOR_REQUEST_ID_H_
+#define DALI_BACKEND_DALI_EXECUTOR_REQUEST_ID_H_
+
+#include <string>
+
+namespace triton::backend::dali {
+
+template<typename T = uint64_t>
+struct RequestId {
+  /**
+   * Prefix increment.
+   */
+  T& operator++() {
+    request_id_str_ = std::to_string(++request_id_);
+    return request_id_;
+  }
+
+
+  /**
+   * Postfix increment.
+   */
+  T operator++(int) {
+    T old = request_id_;
+    operator++();
```

Comment:
Readability nitpick:
```suggestion
    ++*this;
```